server/db/driver/pg/orders.go

```diff
@@ -268,13 +261,40 @@ func (a *Archiver) CancelOrder(lo *order.LimitOrder) error {
 	return a.updateOrderStatus(lo, orderStatusCanceled)
 }
 
-// RevokeOrder updates a LimitOrder with revoked status, which is used for
+// RevokeOrder updates an Order with revoked status, which is used for
 // DEX-revoked orders rather than orders matched with a user's CancelOrder. If
 // the order does not exist in the Archiver, RevokeOrder returns
-// ErrUnknownOrder. To store a new limit order with revoked status, use
-// StoreOrder.
-func (a *Archiver) RevokeOrder(lo *order.LimitOrder) error {
-	return a.updateOrderStatus(lo, orderStatusRevoked)
+// ErrUnknownOrder. This may change orders with status executed to revoked,
+// which may be unexpected.
+func (a *Archiver) RevokeOrder(ord order.Order) (cancelID order.OrderID, timeStamp time.Time, err error) {
+	// Revoke the targeted order.
+	err = a.updateOrderStatus(ord, orderStatusRevoked)
```

Comment:
If we don't want to allow a single order to result in multiple virtual cancel orders in storage and cancel strikes in the auth manager when there are multiple swap revokes involving the order, we could have `Archiver.updateOrderStatus` return the initial status and skip virtual cancel order generate if the order was already revoked, and `RevokeOrder` return values that allow the caller of (`checkInaction`) to skip recording another cancel with the auth manager in this case.  I need to reflect on this some...